app/controllers/posts_controller.rb

```diff
@@ -6,7 +6,7 @@ def index
 
   def show
     @user = User.find(params[:user_id])
-    @post = @user.posts.find(params[:id])
+    @post = Post.find(params[:id])
     @comments = @post.comments
```

Comment:
- [ ] As stated in the requirements, you should try to solve the n+1 queries but your current implementation does not necessarily solve this. Please kindly ensure that you refactor to solve the `n+1` queries.

Hint:
`@posts = @user.posts.includes(:comments)`